### DISCUSSION TITLE
Abbreviation Expansion Design Solutions

### DISCUSSION THREAD

**Discussion by sofia-bg** [UNC-Project-Open-AAC/OS-DPI#99] [Ideas]
At the community meeting the idea of abbreviation expansion was brought up. 

Creating this thread to share ideas and approaches about how to do this in a data driven way. 

My inclination is that you might have a sheet that has two columns, one with the letter sequences and one with the expansions. Trying to think through what the actions would be to have the system recognize when a letter combination was activated in the keyboard, or present in the display, that matches the letter sequence in the sheet and replaces the sequence with the data in the expansion column. 

Will play around and post attempted solutions here, but would be interested in seeing what others come up with if they attempt.

**Comment by gbishop** [comment]
Take a look at the unity simulation to see another approach. The letters could be treated like the symbols in that model. 

But I think treating it just like word prediction should work. Why not just merge your abbreviations into the prediction list? Doesn't that just work?
